File: 3383-TakingMaximumEnergyFromTheMysticDungeon/3383-TakingMaximumEnergyFromTheMysticDungeon.java
// Last Updated: 2/5/2026, 9:21:32 AM
class Solution {
    public int maximumEnergy(int[] energy, int k) {
        int maxEnergy=Integer.MIN_VALUE;
        int n=energy.length;
        int energySequence[] = new int[n];
        for (int x=n-1 ; x>=0 ; x--){
            energySequence[x] = energy[x]+(x+k<n? energySequence[x+k] : 0);
            maxEnergy = Math.max(maxEnergy,energySequence[x]);
        }
        return maxEnergy;
    }
}

File: 3383-TakingMaximumEnergyFromTheMysticDungeon/3383-TakingMaximumEnergyFromTheMysticDungeon.py
# Last Updated: 2/5/2026, 9:21:38 AM
class Solution:
    def maximumEnergy(self, energy: List[int], k: int) -> int:
        maxEnergy = float('-inf')
        energySequence = [0]*len(energy)
        for x in range(len(energy)-1,-1,-1):
            energySequence[x] = energy[x]+(energySequence[x+k] if x+k<len(energy) else 0)
            maxEnergy = max(maxEnergy,energySequence[x])
        return maxEnergy
